### DISCUSSION TITLE
In a middleware: How to determine, which middleware is used on the target route?

### DISCUSSION THREAD

**Discussion by ahoiroman** [Teranode/nuxt-module-alternatives#79] [Q&A]
Hello there,

I'd like to create a little middleware that checks, whether a user has verified the email address. This fact is provided by the user-object coming from the Laravel/Sanctum API.

Now I thought about something like:

1. Create a global middleware, that checks, whether the requested page is visible for authenticated users only
2. Check, whether the user is logged in
3. Check, whether the user has verified the email address

I am using global auth middleware, that protects all routes - excluded those, which have the `guest` middleware defined:

```
definePageMeta({
    auth: 'guest',
});
```

Now I could go and check `to.meta.auth`.

This is my example:

`middleware/userIsVerified.global.ts`:

```ts
export default defineNuxtRouteMiddleware(async (to, from) => {
    console.log("Test", to);
    //if (to.meta.auth) ...check, if the user is verified
    // ...
    navigateTo(to);
})
```

But is this the supposed way to do so or are there better possibilities to check, whether the user authentication is required for a specific page?

**Comment by toniengelhardt** [comment]
Hi @ahoiroman, how did you set up the auth middleware globally? In Nuxt 2 I had this in `nuxt.config.ts`, but it doesn't seem to work in v3: 

```ts
router: {
    base: '/',
    middleware: [
      'auth',
    ],
}
```

**Reply by toniengelhardt** [reply]
Nevermind, answer here: https://github.com/Teranode/nuxt-module-alternatives/discussions/92.
